aws_lambda_powertools/utilities/parser/models/s3.py

```diff
@@ -11,88 +11,319 @@
+            "vlR7PnpV2Ce81l0PRw6jlUpck7Jo5ZsQjryTjKlc5aLWGVHPZLj5NeC6qMa0emYBDXOo6QBU0Wo=",
+        ],
+    )
 
 
 class S3OwnerIdentify(BaseModel):
-    principalId: str
+    principalId: str = Field(
+        description="Amazon customer ID of the bucket owner.",
+        examples=[
+            "A3I5XTEXAMAI3E",
+            "A1YQ72UWCM96UF",
+            "AWS:AIDAINPONIXQXHT3IKHL2",
+        ],
+    )
 
 
 class S3Bucket(BaseModel):
-    name: str
+    name: str = Field(
+        description="Name of the Amazon S3 bucket.",
+        examples=[
+            "lambda-artifacts-deafc19498e3f2df",
+            "example-bucket",
+            "sourcebucket",
+        ],
+    )
     ownerIdentity: S3OwnerIdentify
-    arn: str
+    arn: str = Field(
+        description="The ARN of the Amazon S3 bucket.",
+        examples=[
+            "arn:aws:s3:::lambda-artifacts-deafc19498e3f2df",
+            "arn:aws:s3:::example-bucketarn:aws:s3:::sourcebucket",
+        ],
+    )
 
 
 class S3Object(BaseModel):
-    key: str
-    size: Optional[NonNegativeFloat] = None
-    eTag: Optional[str] = None
-    sequencer: Optional[str] = None
-    versionId: Optional[str] = None
+    key: str = Field(
+        description="The object key (file name) of the S3 object.",
+        examples=[
+            "my-image.jpg",
+            "documents/report.pdf",
+            "logs/2023/01/15/app.log",
+        ],
+    )
+    size: Optional[NonNegativeFloat] = Field(
+        default=None,
+        description="The size of the object in bytes.",
+        examples=[1024, 2048576, 0],
+    )
+    eTag: Optional[str] = Field(
+        default=None,
+        description="The entity tag (ETag) of the object.",
+        examples=[
+            "d41d8cd98f00b204e9800998ecf8427e",
+            "098f6bcd4621d373cade4e832627b4f6",
+        ],
+    )
+    sequencer: Optional[str] = Field(
+        default=None,
+        description="A string representation of a hexadecimal value used to determine event sequence.",
+        examples=[
+            "0A1B2C3D4E5F678901",
+            "005B21C13A6F24045E",
+        ],
+    )
+    versionId: Optional[str] = Field(
+        default=None,
+        description="The version ID of the object (if versioning is enabled).",
+        examples=[
+            "096fKKXTRTtl3on89fVO.nfljtsv6qko",
+            "null",
+        ],
+    )
 
 
 class S3Message(BaseModel):
-    s3SchemaVersion: str
-    configurationId: str
+    s3SchemaVersion: str = Field(
+        description="S3 schema version.",
+        examples=[
+            "1.0",
+        ],
+    )
+    configurationId: str = Field(
+        description="ID of the bucket notification configuration.",
+        examples=[
+            "828aa6fc-f7b5-4305-8584-487c791949c1",
+            "f99fa751-7860-4d65-86cb-10ff34448555",
+            "b1d3a482-96eb-4d3a-abd7-763662a6ba94",
+        ],
+    )
     bucket: S3Bucket
```

Comment:
Looking at other [similar PRs](https://github.com/aws-powertools/powertools-lambda-python/pull/7287/files#diff-85bc4ede1bb54cfaee8c061b057b50f633882dc55dfb9f0ca00cbe4135957b91R151), it seems to me that also complex types should be documented.

I'll amend my PR tonight/tomorrow; so far it only covers basic data types.

Reply:
Yess please!